packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -784,6 +786,37 @@ class FlutterVmService {
     }
   }
 
+  /// Waits for a signal from the VM service that [extensionName] is registered.
+  ///
+  /// Looks at the list of loaded extensions for first Flutter view, as well as
+  /// the stream of added extensions to avoid races.
+  Future<vm_service.IsolateRef> findExtensionIsolate(String extensionName) async {
+    final vm_service.IsolateRef isolateRef = (await getFlutterViews()).first.uiIsolate;
+    try {
+      await service.streamListen(vm_service.EventStreams.kIsolate);
+    } on vm_service.RPCError {
+      // Do nothing, since the tool is already subscribed.
+    }
+
+    try {
```

Comment:
I would prefer if this was structured similar to the original code, in a more clearly imperative manner

Reply:
Done, please take a look again 🙂

Reply:
you need to handle the case where isolates is empty. While you are at it, I would consider what happens if there is more than one uiIsolate.

Reply:
I was under the impression that `getFlutterViews` keeps polling until the isolates are found. Is that not the case here?

Also regarding handling more than one isolate, would it be acceptable to run the querying logic (L813-814) over each `uiIsolate` found?

Reply:
I believe it will return an empty list if the vm service disconnects.

Reply:
Sorry, I don't fully understand enough of the VM service to know what to do when it disconnects. Is this a recoverable error where we want to reconnect and make the same call again, or should we fail with an exception?

Reply:
As discussed over chat, made it throw an exception here, and updated `resident_devtools_handler.dart` to ignore the current `FlutterDevice` when this happens